File: ScandicDevJobApi/ScandicDevJobApi/Models/JobListing.cs
using ScandicDevJobApi.Models.Enums.Company;
using ScandicDevJobApi.Models.Enums.JobListing;

namespace ScandicDevJobApi.Models
{
    public class JobListing
    {
        // General
        public int Id { get; set; }
        public string? Title { get; set; }
        public string? Description { get; set; }
        public User? Owner { get; set; }
        public List<Tag>? Tags { get; set; }
        public string? Category { get; set; }
        public EmploymentType EmploymentType { get; set; }
        public WorkMode WorkMode { get; set; }
        public bool IsPublished { get; set; } // Determines if the job is publicly visible
        public int NumberOfApplicants { get; set; } = 0;

        // Company
        public Company? Company { get; set; }
        public string? ContactEmail { get; set; }
        public string? ApplicationUrl { get; set; }

        // Salary & money
        public string? Currency { get; set; }
        public int? SalaryRangeMin { get; set; }
        public int? SalaryRangeMax { get; set; }

        // Location
        public string? Address { get; set; }
        public string? City { get; set; }
        public string? Country { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }

        //Dates
        public DateTimeOffset? CreatedDate { get; set; } = DateTimeOffset.Now;
        public DateTimeOffset? UpdatedDate { get; set; }
        public DateTimeOffset? JobListingExpiryDate { get; set; }
        public DateTimeOffset? ApplicationDeadline { get; set; }


    }
}

File: ScandicDevFrontEnd/src/lib/types/JobListing.ts
import type { User } from '$lib/types/user';
import type { Tag } from '$lib/types/Tag';
import type { Company } from '$lib/types/Company';

export interface Joblisting {
  id: number;
  title?: string;
  description?: string;
  owner?: User;
  tags?: Array<Tag>;
  category?: string;
  employmentType: EmploymentType;
  workMode: WorkMode;
  isPublished: boolean;
  numberOfApplicants: number;
  logoFileId?: string;

  // Company-related fields
  company?: Company;
  contactEmail?: string;
  applicationUrl?: string;

  // Salary & Money
  currency?: string;
  salaryRangeMin?: number;
  salaryRangeMax?: number;

  // Location
  address?: string;
  city?: string;
  country?: string;
  latitude?: number;
  longitude?: number;

  // Dates
  createdDate?: string;
  updatedDate?: string;
  jobListingExpiryDate?: string;
  applicationDeadline?: string;
}